src/mqtt/mqtt_test.c

```diff
@@ -822,14 +832,30 @@ TEST_TEAR_DOWN( MqttTest )
  */
 TEST( MqttTest, MQTT_Subscribe_Publish_With_Qos_0 )
 {
+    MQTTStatus_t xMQTTStatus;
+    uint32_t entryTime;
+
     /* Subscribe to a topic with Qos 0. */
     TEST_ASSERT_EQUAL( MQTTSuccess, subscribeToTopic(
                            &context, TEST_MQTT_TOPIC, MQTTQoS0 ) );
 
     /* We expect a SUBACK from the broker for the subscribe operation. */
     TEST_ASSERT_FALSE( receivedSubAck );
```

Comment:
After going through the rest, I realize this is just because you are making sure that it hasnt received the message before running the loop - may want to update some comments to make that clear as the comment says We expect a suback yet assert false for it on the next line. Especially when there is a comment below that says We do not expect one and then immediately asserts false for it.

Reply:
Done!